lib/cuckoo/common/virustotal.py

```diff
@@ -168,40 +398,227 @@ def file_scan(self, filepath):
         r = self._request_json(self.FILE_SCAN, data=data, files=files)
         return dict(summary=dict(permalink=r.get("permalink")))
 
+    def detect_platform(self, tokens):
```

Comment:
Rather than using AV signatures, we may consider doing this through simple static analysis of a given file. In that case, it probably makes most sense to move this logic to the following library, where it is easier to unit tests (being a small library and all that): https://github.com/jbremer/sflock.